### PR TITLE
Pipe visualizers

### PR DESCRIPTION
**Pull request by collinlunn**
Pipes now look disconnected, then connect when pipes are placed that connect to them. The base pipe sprite looks disconnected, then PipeConnectorVisualizer layers over a small sprite that covers up the disconnected part of the pipe.

Also, you can set a PipeNode to an arbitrary PipeDirection and the sprite will update correctly.

### CODE REVIEW

File: Content.Client/GameObjects/Components/Atmos/Piping/PipeVisualizer.cs [hunk 1]
```diff
@@ -11,42 +12,42 @@
...
-                Logger.ErrorS("go.ventvisualizer", "Unable to load RSI '{0}'. Trace:\n{1}", rsiPath, e);
+                var rsiPath = SharedSpriteComponent.TextureRoot / rsiString;
+                try
+                {
+                    _pipeRSI = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(rsiPath).RSI;
+                }
+                catch (Exception e)
+                {
+                    Logger.ErrorS($"{nameof(PipeVisualizer)}", $"Unable to load RSI {rsiPath}. Trace:\n{e}");
+                }
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/Atmos/Piping/PipeVisualizer.cs:43]: Same here.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Maybe something for next pr but some stuff (e.g. vents and scrubbers) are triggering EnsureComponentWarn for PipeNetDeviceComponent.

File: Content.Client/GameObjects/Components/Atmos/Piping/PipeConnectorVisualizer.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            {
+                var rsiPath = SharedSpriteComponent.TextureRoot / rsiString;
+                try
+                {
+                    _connectorRsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(rsiPath).RSI;
+                }
+                catch (Exception e)
+                {
+                    Logger.ErrorS($"{nameof(PipeConnectorVisualizer)}", $"Unable to load RSI {rsiPath}. Trace:\n{e}");
+                }
```
[the remark this answers is not in the document]
Comment (collinlunn) [reply Content.Client/GameObjects/Components/Atmos/Piping/PipeConnectorVisualizer.cs:44]: The specific exception, `FileNotFoundException`, seems to violate the sandbox. Not sure how to handle that.

[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Client/GameObjects/Components/Atmos/Piping/PipeConnectorVisualizer.cs:44]: Does TryGetResource instead?

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**